test: add tests to date_histogram with and without offset and timezone

salvatore-campagna opened the pull request:
The first of these test was meant to reproduce issue #56305. Anyway the result looks correct to me.
The second test is there to have the same test but without the `offset`.

salvatore-campagna committed:
test: add tests to date_histogram with and without offset and timezone change

elasticmachine commented:
Pinging @elastic/es-analytics-geo (Team:Analytics)

salvatore-campagna commented on rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/360_date_histogram.yml:
```diff
@@ -356,3 +385,58 @@ setup:
+---
+"Daylight with offset date_histogram test":
+  - skip:
+      version: "- 8.2.99"
```
I know this is incorrect...I will try to see if I can find when this was fixed (assuming the test results are correct)

not-napoleon replied:
As noted, I think 7.17 is good enough.

elasticmachine commented:
Pinging @elastic/clients-team (Team:Clients)

not-napoleon commented on rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/360_date_histogram.yml:
```diff
@@ -356,3 +385,58 @@ setup:
+
+---
+"Daylight without offset date_histogram test":
+  - skip:
```
Without DST, this probably doesn't need a skip version at all.

not-napoleon reviewed (approved):
LGTM

nik9000 reviewed (approved):
OK then. I don't know how or when we fixed this, but yay.

salvatore-campagna committed:
test: change skip version to 7.17.0

The issue with date histogram and offsets was fixed in some version
before 7.17.0.
